app/components/profile_picture/profile_picture.js

```diff
@@ -106,6 +107,13 @@ export default class ProfilePicture extends PureComponent {
         }
     }
 
+    componentDidUpdate(prevProps) {
+        // Remove the profile image from the UI before removing from server
+        if (this.props.profileImageRemove !== prevProps.profileImageRemove) {
+            this.setImageURL(null);
```

Comment:
This is considered bad form since `setImageURL` calls `setState` and will trigger a second re-render after this one has just completed. What we can do instead is clear the the value of `profileImageUri` in the store so that we don't need to add an extra prop for it.

What we can do instead is have `removeProfileImage` also dispatch `setProfileImageUri` to clear it from the store immediately.
```
export function removeProfileImage(user) {
    return async (dispatch) => {
        dispatch(setProfileImageUri(''));
        return await dispatch(setDefaultProfileImage(user));
    };
}
```

Reply:
Actually, while that would be my preferred way of doing it, this component's `componentWillReceiveProps` is written in a way that doesn't allow this to because of the `nextProps.profileImageUri !== ''` [here](https://github.com/mattermost/mattermost-mobile/blob/9763eeadda02161c025355646731cefef0aa8f6a/app/components/profile_picture/profile_picture.js#L80), although removing that will cause other problems.

I'm fine leaving you changes as-is, but if you're interested in doing some cleanup on this component, I can elaborate some more on that.

Reply:
I'm happy to do some house-cleaning here @hmhealey 🏚